internal/process/provisioning/kyma_append_modules_test.go

```diff
@@ -0,0 +1,238 @@
+package provisioning
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/fixture"
+	"github.com/kyma-project/kyma-environment-broker/internal/ptr"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	technicalNameBtpManager         = "btp-operator"
+	technicalNameKeda               = "keda"
+	givenKymaTemplateWithModules    = "given_kyma_with_modules.yaml"
+	givenKymaTemplateWithoutModules = "given_kyma_without_modules.yaml"
+	kymaExpectedNamespace           = "kyma-system"
+	kymaAppendModulesTestAssets     = "kyma_append_modules_tests_assets"
+)
+
+func execTest(t *testing.T, parameters *internal.ModulesDTO, in, out string) {
+	db := storage.NewMemoryStorage()
+	operation := fixture.FixOperation(uuid.NewString(), uuid.NewString(), internal.OperationTypeProvision)
+	operation.KymaTemplate = internal.GetFileWithTest(t, fmt.Sprintf("%s/%s", kymaAppendModulesTestAssets, in))
+	expectedKymaTemplate := internal.GetFileWithTest(t, fmt.Sprintf("%s/%s", kymaAppendModulesTestAssets, out))
+	operation.ProvisioningParameters.Parameters.Modules = parameters
+	db.Operations().InsertOperation(operation)
+	svc := NewKymaAppendModules(db.Operations())
+
+	// when
+	op, backoff, err := svc.Run(operation, logrus.New())
+	assert.Zero(t, backoff)
+	assert.Equal(t, kymaExpectedNamespace, op.KymaResourceNamespace)
+	assert.YAMLEq(t, expectedKymaTemplate, op.KymaTemplate)
+	require.NoError(t, err)
+}
+
+// when given kyma template without any default modules set
+
+func TestKymaAppendModulesWithEmptyDefaultOnes1_Run(t *testing.T) {
+	params := func() *internal.ModulesDTO {
+		modules := &internal.ModulesDTO{}
+		modules.List = make([]*internal.ModuleDTO, 0)
+		m1 := internal.ModuleDTO{
+			Name:                 technicalNameBtpManager,
+			CustomResourcePolicy: internal.CreateAndDelete,
+		}
+		m2 := internal.ModuleDTO{
+			Name:    technicalNameKeda,
+			Channel: internal.Regular,
+		}
+		modules.List = append(modules.List, &m1, &m2)
+		return modules
+	}
+	execTest(t, params(), givenKymaTemplateWithoutModules, "kyma_template_output_1.yaml")
+}
+
+func TestKymaAppendModulesWithEmptyDefaultOnes2_Run(t *testing.T) {
```

Comment:
So we test adding two modules to empty list and in the next case we test adding one module.
What is the rationale?